docs/rules/no-inline-assertions.md

```diff
@@ -0,0 +1,19 @@
+# Ensure no assertions are called from inline functions
+
+Prevent assertions being called from an inline function, to make it clear that it does not return.
```

Comment:
This is repeating the title too much. It should also use the wording "test implementation".